### PR TITLE
OAuth

### LINKED ISSUE #9: login with oneauth
Let's consider leads tracker frontend is on lt.com and the backend API is available on lt.com/api/

#### This part is done by frontend   
 - User goes to https://lt.com/login (which is simply a frontend route)
 - He is redirected to `http://localhost:3838/oauth/authorize?response_type=code&client_id=OUR_CLIENT_ID_ON_ONEAUTH&redirect_uri=http://lt.com/login/callback`
 - He logs into oneauth and comes back to `http://lt.com/login/callback?code=GRANT_CODE_WE_GET`

#### What we have to handle in backend    
 1. We send a POST API request to `lt.com/api/auth`, with body = `{code: GRANT_CODE_WE_GET}`
 2. Using the code we got, we send a request to `http://localhost:3838/oauth/token` with following body

    {
         "client_id" : OUR_CLIENT_ID_ON_ONEAUTH,
         "redirect_uri" : "http://lt.com/login/callback",
         "client_secret" : "OUR_CLIENT_SECRET_ON_ONEAUTH",
         "grant_type" : "authorization_code",
         "code"  : "GRANT_CODE_WE_GET"
     }
 3. In the response we get `{token: AUTH_TOKEN_WE_RECEIVE}`
 4. We will create a new token ourselves (use uid2). In the db we will create a table with map of OUR_TOKEN: GRANT_CODE_WE_GET
 5. To the frontend we send (in response to the POST request on 1.) this OUR_TOKEN we create. 

We will need a table as such - 

| id | oneauth_id | lt_token | oneauth_token |
|----|------------|----------|---------------|
| 1  | 35         | abc      | pqr           |
| 2  | 87         | def      | xyz           |  

User 1 has user_id 35 on oneauth. User 2 has user_id 87 on oneauth.   
To our frontend we only give the **abc** or **def** token. The oneauth oauth token, that is 
**_pqr_** and **_xyz_** will not be exposed to the frontend and we keep it only in our db.

 - When we need profile data a user, we first find the token in his request. 
 - Let us consider the token is _def_ 
 - We look up the table and find it is user 2.
 - We use his oneauth token, that is _**xyz**_ to send request to `https://account.codingblocks.com/api/user/me` and get his profile data from oneauth.

### PR DESCRIPTION
**Pull request by bhavyaagg**
Fixes #9

### CODE REVIEW

File: src/db/models.ts [hunk 1]
```diff
@@ -51,9 +51,19 @@ const Lead = db.define('lead', {
    , name: Sequelize.STRING
    }
    */
-  , status: Sequelize.STRING
+  , status: {type: Sequelize.ENUM('New','Interested','Uninterested','Enrolled','Untracked'), allowNull: false, defaultValue:'New'}
```
Comment (championswimmer) [inline src/db/models.ts]: use all smallcase in the enum options
new, interested . . .

File: src/routers/api/auth.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  secrets = require('./../../../secrets.json');
+} catch (e) {
+  console.error('Create your own secrets file lazybones');
+  secrets = require('./../../../secret-sample.json');
+}
+
+
+router.post('/', function (req, res) {
+  // TODO: Oneauth ID
+  // console.log(1)
```
Comment (championswimmer) [inline src/routers/api/auth.js]: remove comment

File: public_html/index.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
```
Comment (championswimmer) [inline public_html/index.html]: what is the purpose of this file ?

**Review by championswimmer — CHANGES_REQUESTED** [review]

File: public_html/index.html (hunk 3 above)
Comment (bhavyaagg) [reply public_html/index.html]: Sorry forgot to remove it. It was just for testing the oauth.

**Comment by bhavyaagg** [comment]
@championswimmer Sir i have updated the pull request.

**Review by championswimmer — APPROVED** [review]

**Pull request merged**